homeassistant/components/prometheus.py

```diff
@@ -0,0 +1,201 @@
+"""
+Support for Prometheus metrics export
+
+For more details about this component, please refer to the documentation at
+https://home-assistant.io/components/prometheus/
+"""
+import asyncio
+import logging
+
+from aiohttp import web
+
+from homeassistant.components.http import HomeAssistantView
+from homeassistant.const import (ATTR_HIDDEN, EVENT_STATE_CHANGED,
+                                 TEMP_CELSIUS, TEMP_FAHRENHEIT)
+from homeassistant import core as ha
+from homeassistant.helpers import state as state_helper
+
+_LOGGER = logging.getLogger(__name__)
+
+ATTR_PROMETHEUS_HIDDEN = 'prometheus_hidden'
+
+REQUIREMENTS = ['prometheus_client==0.0.18']
+
+DOMAIN = 'prometheus'
+DEPENDENCIES = ['http']
+
+
+def setup(hass, config):
+    import prometheus_client
+
+    hass.http.register_view(PrometheusView(hass, prometheus_client))
+
+    metrics = Metrics(prometheus_client)
+
+    hass.bus.listen(EVENT_STATE_CHANGED, metrics.handle_event)
+
+    return True
+
+
+class Metrics:
+    def __init__(self, prometheus_client):
+        self.prometheus_client = prometheus_client
+        self._metrics = {}
+
+    def handle_event(self, event):
+        """Listen for new messages on the bus, and add them to Prometheus"""
+        state = event.data.get('new_state')
+        if state is None:
+            return
+
+        _LOGGER.info("Handling state update for %s", state.entity_id)
+
+        if (state.attributes.get(ATTR_HIDDEN) or
+                state.attributes.get(ATTR_PROMETHEUS_HIDDEN)):
```

Comment:
I think that it makes more sense if this would be implemented as config options to the component. See the history component for an example: https://github.com/home-assistant/home-assistant/blob/dev/homeassistant/components/history.py#L31-L40